### ISSUE TITLE
Laravel4 Functional Test Failing Due To A Thrown Exception

### ISSUE THREAD

**Issue by weblee** [Codeception/Codeception#1227]
I am running a simple functional test and as part of that test I am running a validation which  fails and throws an exception. 

If I catch the exception within the controller itself I can get the test to pass. But I am catching the exception globally in the app/start/global.php

Code Example Passing.

``` php
<?php

use Acme\Forms\SigInForm;

class SessionsController extends \BaseController {

    private $sigInForm;

    function __construct(SigInForm $sigInForm)
    {
        $this->sigInForm = $sigInForm;
    }

    public function store()
    {
          $input = Input::only('email', 'password');

        try{
            $this->sigInForm->validate($input);
        }catch(FormValidationException $e)
        {
            Redirect::back()->withInput();
        }

          if(Auth::attempt($input))
          {
            Flash::success('Welcome Back');
            return Redirect::intended('extranet');
          }

    }
}
```

If I remove the try catch and let the app/start/global.php catch the Exception its does not. I just get the Exception being displayed eg:

```
Time: 2.12 seconds, Memory: 19.75Mb

There was 1 error:

---------
1) Failed to attempt to login inGuestSignInCept (/home/domain/tests/functional/GuestSignInCept.php)
Couldn't sign in :
Exception: FormValidationException failed

Scenario Steps:
2. I sign in
1. As a a guest

#1  /home/airportparking/app/controllers/SessionsController.php:45
```

This is the global.php file where I handle the exception

``` php
App::error(function(Acme\Validation\FormValidationException $exception, $code){
    return Redirect::back()->withInput()->withErrors($exception->getErrors());
});
```

I suppose it would be good if there was a method to catch the exception or record it ie,

``` php
$I->seeException(...)
```

**Comment by supercoffee** [comment]
I had the same problem.  To debug the issue, I var_dumped the registered error handlers in Laravel `Illuminate\Exception\Handler` class where exception handlers are registered during the test run.  There were none.  In my case, I had created a separate `errors.php` to register my error handlers and did a `require_once __DIR__.'/errors.php';` from my `start/global.php` file. 

Turns out that the Codeception Laravel module loads up the framework from the `global.php` file, but since I had used `require_once` to execute my error handlers file, the error handlers file was only be executed once per test suite run. 

I realize that my case and solution might not be applicable to anyone else here, but I though I'd share my debugging experience on this issue. 

In short, check your error handlers to make sure they are registered during the test run.  Also make sure that Laravel is not running in the `testing` environment because it will not call your error handlers there.
